tokio/src/io/async_fd.rs

```diff
@@ -508,6 +508,32 @@ impl<T: AsRawFd> AsyncFd<T> {
     pub async fn writable_mut<'a>(&'a mut self) -> io::Result<AsyncFdReadyMutGuard<'a, T>> {
         self.readiness_mut(Interest::WRITABLE).await
     }
+
+    /// Waits for the file descriptor to become priority ready, returning a
+    /// [`AsyncFdReadyGuard`] that must be dropped to resume priority-readiness
+    /// polling.
+    ///
+    /// This method takes `&self`, so it is possible to call this method
+    /// concurrently with other methods on this struct. This method only
+    /// provides shared access to the inner IO resource when handling the
+    /// [`AsyncFdReadyGuard`].
+    #[allow(clippy::needless_lifetimes)] // The lifetime improves rustdoc rendering.
+    #[cfg(any(target_os = "linux", target_os = "android"))]
```

Comment:
We should add the information about supported targets to the docs, please add this attribute here and in other places as well.
```suggestion
    #[cfg(any(target_os = "linux", target_os = "android"))]
    #[cfg_attr(docsrs, doc(cfg(any(target_os = "linux", target_os = "android"))))]
```